ci-operator/step-registry/cucushift/installer/check/aws/private/cucushift-installer-check-aws-private-commands.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+REGION="${LEASED_RESOURCE}"
+INFRA_ID=$(jq -r '.infraID' ${SHARED_DIR}/metadata.json)
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+ret=0
+
+#Check there's no external load balancer created in this cluster
+CLUSTER_LB_LIST=$(mktemp)
+
+for lb in $(aws --region ${REGION} elbv2 describe-load-balancers | jq -r '.LoadBalancers[].LoadBalancerArn'); do
+    aws --region ${REGION} elbv2 describe-tags --resource-arns "$lb" | jq --arg infra "kubernetes.io/cluster/${INFRA_ID}" -ce '.TagDescriptions[].Tags[] | select( .Key == "\($infra)" and .Value == "owned")' && echo "$lb" >> "${CLUSTER_LB_LIST}"
+done
+
+for cluster_lb in $(cat $CLUSTER_LB_LIST); do 
+    LB_TYPE=$(aws --region ${REGION} elbv2 describe-load-balancers --load-balancer-arns ${cluster_lb} | jq -r '.LoadBalancers[].Scheme')
+    if [[ "${LB_TYPE}" == "internal" ]]; then
+        echo "Pass: LB ${cluster_lb} is internal."
+    else
+        echo "Error: LB ${cluster_lb} is not internal, the LB type is ${LB_TYPE}"
+        ret=$((ret + 1))
+    fi
+done
+
+
+#No public Route 53 DNS records that matches the baseDomain for the cluster
+if [[ -z ${BASE_DOMAIN} ]]; then
+  echo "Error: BASE_DOMAIN is not set, exit."
+  exit 1
+fi
+
+PUBLIC_ZONE_ID=$(aws route53 list-hosted-zones-by-name | jq --arg name "${BASE_DOMAIN}." -r '.HostedZones | .[] | select(.Name=="\($name)") | .Id' | awk -F / '{printf $3}')
+
+CLUSTER_PUBLIC_DNS=$(aws route53 list-resource-record-sets --hosted-zone-id=${PUBLIC_ZONE_ID} --output json | jq '.ResourceRecordSets[].Name' |grep "${CLUSTER_NAME}.${BASE_DOMAIN}")
```

Comment:
```suggestion
CLUSTER_PUBLIC_DNS=$(aws route53 list-resource-record-sets --hosted-zone-id=${PUBLIC_ZONE_ID} --no-paginate --output json | jq '.ResourceRecordSets[].Name' |grep "${CLUSTER_NAME}.${BASE_DOMAIN}")
```

Reply:
Seems the "--no-paginate" can't work in this case, with adding "--no-paginate", the command can't filter out the expected keywords.

```
[root@preserve-gpei-worker ~]# aws route53 list-resource-record-sets --hosted-zone-id=${PUBLIC_ZONE_ID} --no-paginate --output json | jq '.ResourceRecordSets[].Name' |grep "abhbaner-aw.qe.devcluster.openshift.com"

[root@preserve-gpei-worker ~]# aws route53 list-resource-record-sets --hosted-zone-id=${PUBLIC_ZONE_ID}  --output json | jq '.ResourceRecordSets[].Name' |grep "abhbaner-aw.qe.devcluster.openshift.com"
"api.abhbaner-aw.qe.devcluster.openshift.com."
"\\052.apps.abhbaner-aw.qe.devcluster.openshift.com."
```

Reply:
My fault, I mis-understanding the parameter:
```
       --no-paginate (boolean)

       Disable  automatic pagination. If automatic pagination is disabled, the
       AWS CLI will only make one call, for the first page of results.
```

Sounds like we should NOT use the options if we want to get all pages of results, so we also need to revert the change on `elbv2 describe-load-balancers` command. Sorry for that.